Library/Homebrew/os/mac/version.rb

```diff
@@ -33,7 +34,7 @@ def self.from_symbol(sym)
         new(str, arch: arch)
       end
 
-      sig { params(value: T.any(String, Symbol)).returns(T::Array[String]) }
+      sig { params(value: T.any(String, Symbol)).returns(T::Array[T.nilable(String)]) }
```

Comment:
Is this nilable? Could it not be?

Reply:
Yes, the second element, `arch`, can be nil

We could also do:

```rb
returns(T.any([], [String, T.nilable(String)]))
```

but I'm not sure if that's an improvement

Reply:
Yeh, that feels like a mild improvement 👍🏻